tests/func/test_run.py

```diff
@@ -611,93 +611,64 @@ def test_cwd_is_ignored(self):
-            repo_dir.CODE, repo_dir.FOO, self.out_file
-        )
-        self.deps = [repo_dir.FOO, repo_dir.CODE]
-        self.outs = [self.out_file]
-        self.overwrite = False
-        self.ignore_build_cache = False
-        self.dvc_repo = dvc_repo
-        self.stage = None
-
-    def run(self):
-        self.stage = self.dvc_repo.run(
-            cmd=self.cmd,
-            fname=self.stage_file,
-            overwrite=self.overwrite,
-            ignore_build_cache=self.ignore_build_cache,
-            deps=self.deps,
-            outs=self.outs,
-        )
-        return self.stage
+def test_rerun_deterministic(tmp_dir, dvc, run_copy):
+    tmp_dir.gen("foo", "foo content")
 
+    assert run_copy("foo", "out") is not None
+    assert run_copy("foo", "out") is None
 
-@pytest.fixture
-def deterministic_run(dvc_repo, repo_dir):
-    run_base = DeterministicRunBaseFixture(repo_dir, dvc_repo)
-    run_base.run()
-    yield run_base
 
+def test_rerun_deterministic_ignore_cache(tmp_dir, dvc, run_copy):
+    tmp_dir.gen("foo", "foo content")
 
-def test_run_deterministic(deterministic_run):
-    deterministic_run.run()
+    assert run_copy("foo", "out") is not None
+    assert run_copy("foo", "out", ignore_build_cache=True) is not None
 
 
-def test_run_deterministic_overwrite(deterministic_run):
-    deterministic_run.overwrite = True
-    deterministic_run.ignore_build_cache = True
-    deterministic_run.run()
+def test_rerun_callback(tmp_dir, dvc):
+    def run_callback():
+        out = "out"
+        return dvc.run(
+            cmd=(
+                "python -c 'from pathlib import Path;"
+                'Path("{}").write_text("content")\''
+            ).format(out),
+            outs=["out"],
+            deps=[],
+            overwrite=False,
+        )
 
+    assert run_callback() is not None
 
-def test_run_deterministic_callback(deterministic_run):
-    with deterministic_run.stage.repo.lock:
-        deterministic_run.stage.remove()
-    deterministic_run.deps = []
-    deterministic_run.run()
     with mock.patch("dvc.prompt.confirm", return_value=True):
-        assert deterministic_run.run()
-
+        assert run_callback() is not None
 
-def test_run_deterministic_changed_dep(deterministic_run, repo_dir):
-    os.unlink(repo_dir.FOO)
-    shutil.copy(repo_dir.BAR, repo_dir.FOO)
-    with pytest.raises(StageFileAlreadyExistsError):
-        deterministic_run.run()
 
+def test_rerun_changed_dep(tmp_dir, dvc, run_copy):
+    tmp_dir.gen("foo", "foo content")
+    assert run_copy("foo", "out") is not None
 
-def test_run_deterministic_changed_deps_list(deterministic_run, repo_dir):
-    deterministic_run.deps = [repo_dir.BAR, repo_dir.CODE]
+    tmp_dir.gen("foo", "changed content")
     with pytest.raises(StageFileAlreadyExistsError):
-        deterministic_run.run()
+        run_copy("foo", "out", overwrite=False)
 
 
-def test_run_deterministic_new_dep(deterministic_run, repo_dir):
```

Comment:
In my opinion `test_run_deterministic_new_dep` `test_run_deterministic_remove_dep` `test_run_deterministic_changed_cmd` `test_run_deterministic_changed_deps_list` were testing the same use case: when we run the same command, but for some reason, change our stage file. That is why I removed them and introduced `test_rerun_changed_stage`